lib/webauthn/signature_verifier.rb

```diff
@@ -9,8 +9,9 @@ module WebAuthn
   class SignatureVerifier
     class UnsupportedAlgorithm < Error; end
 
-    def initialize(algorithm, public_key)
+    def initialize(algorithm, supported_algorithms, public_key)
       @algorithm = algorithm
+      @supported_algorithms = supported_algorithms
```

Comment:
Smells a bit that a class has the responsibility of validating the relationship between two arguments that come from the outside.

I can forsee the receiver potentially telling the caller: "if you know which are the supported algorithms, can you please just make sure you call me with a supported algorithm all the time?" :smile: